File: backend/src/main/java/com/paperai/model/entity/Task.java
package com.paperai.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * Agent 任务实体
 *
 * @author: ch
 * @date 2026年05月11日
 */
@Data
@TableName("task")
public class Task implements java.io.Serializable {
    private static final long serialVersionUID = 1L;

    /** 任务ID */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 关联论文ID */
    private Long paperId;

    /** Agent 角色编码 */
    private String agentRole;

    /** 关联版本号 */
    private Integer versionNo;

    /** 执行顺序 */
    private Integer sortOrder;

    /** 任务描述 */
    private String description;

    /** 任务输入 */
    private String inputData;

    /** 任务输出 */
    private String outputData;

    /** 状态（PENDING/IN_PROGRESS/COMPLETED/FAILED） */
    private String status;

    /** 耗时（毫秒） */
    private Long durationMs;

    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /** 完成时间 */
    private LocalDateTime completedAt;
}

File: backend-py/app/models/task.py
from datetime import datetime
from sqlalchemy import String, Text, Integer, BigInteger, DateTime, func, Index
from sqlalchemy.orm import Mapped, mapped_column
from .base import Base


class Task(Base):
    __tablename__ = "task"
    __table_args__ = (
        Index("ix_task_paper_id", "paper_id"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    paper_id: Mapped[int | None] = mapped_column(BigInteger)
    agent_role: Mapped[str] = mapped_column(String(30), nullable=False)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    version_no: Mapped[int] = mapped_column(Integer, default=0)
    description: Mapped[str | None] = mapped_column(String(500))
    input_data: Mapped[str | None] = mapped_column(Text)
    output_data: Mapped[str | None] = mapped_column(Text)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="PENDING")
    duration_ms: Mapped[int | None] = mapped_column(BigInteger)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), nullable=False
    )
    completed_at: Mapped[datetime | None] = mapped_column(DateTime)
